### DISCUSSION TITLE
Corellation

### DISCUSSION THREAD

**Discussion by gilang2287** [teamalgoritma/community#530] [Q&A] [answered]
Selamat Pagi, 
saya mohon bantuannya untuk  fungsi cor() untuk menggunakannya seperti apa ya?
krn data yg saya peroleh itu tidak bisa digunakan pada fungsi cor(), dan knp baru bisa digunakan apabila kita membuat data frame terlebih dahulu secara manual?
![image](https://user-images.githubusercontent.com/123464444/216734496-c1e48a84-17ae-46cb-b50f-ae6572f40d3f.png)
![image](https://user-images.githubusercontent.com/123464444/216734629-4acc2035-6b33-46d3-8209-a5d7e835262e.png)

**Comment by muhrisman** [comment, marked as answer]
Izin bantu jawab
Fungsi cor() atau fungsi fungsi lain yang ada di R bisa kita gunakan dengan syarat object yang mau kita cari harus tersimpan dulu di environment. Berdasarkan screenshot tersebut saya lihat bapak belum menyimpan object "total" karena mungkin terlewat untuk dijalankan.
Silahkan bapak jalankan dulu kodingan pada baris yang menyimpan object "total".

Cor() baru bisa digunakan karena kita butuh informasi tambahan "Mau mencari korelasi di data yang mana?" oleh karena itu data frame/object yang mau kita cari korelasinya harus ada terlebih dahulu.

Best Regards,
M Risman

**Reply by gilang2287** [reply]
terima kasih informasinya, tsetelah saya coba masih belum bisa di jalankan, mohon arahannya

![image](https://user-images.githubusercontent.com/123464444/216752203-2909e3f9-57e6-49db-bcce-9061b99a1245.png)

**Reply by muhrisman** [reply]
itu objectnya belum ada di environtment pak, bapak lihat di bagian environtment juga tidak ada object total.

Coba bapak telusuri kodingan bapak yang ada " total <-  " kalau sudah muncul dijalankan chunknya, Kalau tidak ada bapak buat terlebih dahulu atau dihapus saja jika memang bapak tidak sedang mencari cor(quantity, total)

**Reply by gilang2287** [reply]
maaf pak utk yg akan di cor() tsb datanya sdh ada di supermarket, apakah hrs di buat environtment itu sdri krn sdh ada total di supermarket tsb?

terima kasih

**Reply by muhrisman** [reply]
Kalau memang variabelnya ada di supermarket coba gunakan supermarket$nama variabel pak

**Reply by gilang2287** [reply]
baik akan saya coba

**Comment by ELITSZORD** [comment]
Dear @gilang2287 

![Ans cov](https://user-images.githubusercontent.com/114390619/216795641-93750c09-1961-43dd-8045-9076987e28c4.png)

Dimana jika menggunakan data frame harus di mentioned series/kolomnya pak dengan $, dan untuk operasi Correlation, Variance and Covariance sebaiknya input datanya adalah tipe data numeric.

Thanks.

**Reply by kevbow** [reply]
Dear @gilang2287,

Sebagai pelengkap atas informasi yang disampaikan oleh @ELITSZORD, berikut saya lampirkan kode yang beliau gunakan dengan sedikit adjustment sesuai kasus yang Anda angkat di awal:
```r
cor(supermarket$quantity, supermarket$total)
```
Secara jelasnya kode di atas mengatakan bahwa kita akan mengeluarkan nilai korelasi berdasarkan kolom quantity pada data supermarket dengan kolom total pada data supermarket.
Anda tidak bisa langsung mengakses nama kolom saja karena nama kolom tidak tersimpan sebagai object pada environment. Hal ini menyebabkan R tidak menemukan nilai tersebut. Maka dari itu kita menggunakan subsetting kolom pada dataframe dengan keterangan kolom yang ingin diambil nilainya (contoh: `dataframe$nama_kolom`).
Demikian yang dapat saya sampaikan, semoga membantu.
